packages/bookshelf-include-count/lib/bookshelf-include-count.js

```diff
@@ -157,8 +226,71 @@ module.exports = function (Bookshelf) {
             });
 
             return attrs;
-        }
+        },
+
+        /**
+         * Instead of adding the counts in .fetch and .fetchAll,
+         * we need to do it in sync because Bookshelf doesn't call fetch for eagerRelations
+         * E.g. when trying to load counts on replies.count.likes, we wouldn't get an opportunity to load the counts on the replies relation.
+         */
+        sync: function (options) {
+            if (!options.method || (options.method !== 'insert' && options.method !== 'update')) {
+                this.addCounts.apply(this, arguments);
+            }
+
+            if (_debug.enabled('ghost-query')) {
+                debug('QUERY', this.query().toQuery());
+            }
```

Comment:
I've asked in slack about this pattern, it doesn't make sense to me but I appreciate you're following the existing convention